src/txn/txn_ckpt.c

```diff
@@ -850,23 +850,28 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
 
     /*
      * Get a history store dhandle. If the history store file is opened for a special operation this
-     * will return EBUSY which we treat as an error.
+     * will return EBUSY which we treat as an error. In scenarios where the history store is not
+     * part of the metadata file (performing recovery on backup folder where no checkpoint
+     * occurred), this will return ENOENT which we ignore and continue.
      */
-    WT_ERR(__wt_session_get_dhandle(session, WT_HS_URI, NULL, NULL, 0));
-    hs_dhandle = session->dhandle;
+    ret = __wt_session_get_dhandle(session, WT_HS_URI, NULL, NULL, 0);
```

Comment:
Can you simplify the flow here, to have:

`WT_ERR_ERROR_OK(ret, ret = __wt_session_get_dhandle(session, WT_HS_URI, NULL, NULL, 0));`

Then remove the `if (ret == 0) {`?

Reply:
Fixed.